python/cudf/cudf/core/scalar.py

```diff
@@ -4,14 +4,12 @@
 
 import numpy as np
 import pyarrow as pa
-from pandas._libs.missing import NAType as pd_NAType
 
 import cudf
-from cudf.core.column.column import ColumnBase
+from cudf.api.types import is_scalar
 from cudf.core.dtypes import ListDtype, StructDtype
-from cudf.core.index import BaseIndex
+from cudf.core.missing import NA
```

Comment:
Since `cudf` is already imported, do we want to use `cudf.NA` rather than import this name? I see [quite](https://github.com/rapidsai/cudf/blob/027c34aefbf8a5abf5394da15a7b6f1dcc63b06c/python/cudf/cudf/core/column/struct.py) [a](https://github.com/rapidsai/cudf/blob/5d2e206ee98db48396c87d5f1278c50ba265e547/python/cudf/cudf/core/column/numerical_base.py) [few](https://github.com/rapidsai/cudf/blob/1f8a03e69704562dfac38de40b7172650280c6ea/python/cudf/cudf/core/column/lists.py) [files](https://github.com/rapidsai/cudf/blob/1a457efc019ee06bf11d350485cee12087db9d6e/python/cudf/cudf/core/dataframe.py) that use `cudf.NA` rather than importing `NA` separately. I don't think any other files use this convention.

Reply:
This is a really great question. We use `cudf.NA` almost everywhere, but should we? In some ways it feels like it might fit the "onion model" a bit better if `scalar.py` didn't mention the top level `cudf` namespace anywhere. 

I think there's probably plenty of places I have used `cudf.NA` throughout the codebase as a convenience since those places already have `cudf` imported, but I realize now that it has the negative consequence of further entrenching the need for `import cudf` in those places, which is part of what I think gives rise to these import issues.

Is this way off base? I am happy to put in a PR that addresses this for `NA` across the codebase as a follow up if that's what we think should happen, and just imports from `missing`.

Reply:
thanks @vyasr I think that lines up with my understanding. I am going to leave this as-is for now and look into making the change everywhere. I'll leave this thread open though and let @bdice resolve if this conclusion seems satisfactory.

Reply:
Removing `import cudf` sounds fine to me! Could we make an issue to document the plan before resolving this? (I would do it but I’m on mobile at the moment.)

Reply:
I raised https://github.com/rapidsai/cudf/issues/10820 regarding this. I'll tackle this for `NA` first after this PR is merged, after which it should be importable from `missing`. From there I'll start probing what it takes to address this more generally.